pkg/backup/resource_backupper.go

```diff
@@ -174,37 +176,59 @@ func (rb *defaultResourceBackupper) backupResource(
 	)
 
 	var namespacesToList []string
-	if resource.Namespaced {
-		namespacesToList = getNamespacesToList(rb.namespaces)
-	} else {
-		namespacesToList = []string{""}
-	}
-	for _, namespace := range namespacesToList {
-		resourceClient, err := rb.dynamicFactory.ClientForGroupVersionResource(gv, resource, namespace)
+	// If processing the Namespace resource specifically and you've specified to include only some,
+	// fetch them individually instead of a full list of all in the cluster:
+	if resource.Kind == "Namespace" &&
+		!(len(rb.namespaces.GetIncludes()) == 0 || rb.namespaces.GetIncludes()[0] == "*") {
```

Comment:
This could be written in the positive as `(len(rb.namespaces.GetIncludes()) >= 1 && rb.namespaces.GetIncludes()[0] != "*")`. Maybe a bit easier to read?